crates/router/src/core/revenue_recovery/types.rs

```diff
@@ -507,12 +605,14 @@ impl Action {
                     .await;
 
                         // unlocking the token
-                        let _unlock_the_connector_customer_id = storage::revenue_recovery_redis_operation::RedisTokenManager::unlock_connector_customer_status(
+                        if intent_status == common_enums::IntentStatus::Succeeded {
```

Comment:
previously unlock is a doesn't have conditional based unlocking, why now this needs to be conditional here?

And this is for Succeeded we have same for Failed, this logic is not clear.